src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -714,12 +714,16 @@ async def _schedule_single_node_session(
             raise GenericBadRequest(
                 "Cannot assign multiple kernels with different architectures' single node session",
             )
+        if not requested_architectures:
+            raise GenericBadRequest("Requested session has no architecture information")
         requested_architecture = requested_architectures.pop()
         compatible_candidate_agents = [
             ag for ag in candidate_agents if ag.architecture == requested_architecture
         ]
 
         try:
+            if not candidate_agents:
+                raise InstanceNotAvailable(extra_msg="No agents are registered with the manager")
```

Comment:
👍
"No agents are available for scheduling" error msg would be more comprehensible

Reply:
Could you check the error message again? The sentence "Registered with the manager" can confuse clients

Reply:
Could you check the error message again? The sentence "Registered with the manager" can confuse clients